signal/handler/socket.go

```diff
@@ -32,7 +35,10 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		return
 	}
+
+	h.metrics.DecrementWebSocketConnections()
 	defer func(conn *websocket.Conn) {
+		h.metrics.DecrementWebSocketConnections()
```

Comment:
How do you think about publish event that socket disconnection, and subscribe it in Metric?